packages/gatsby-plugin-sharp/src/utils.js

```diff
@@ -124,19 +139,22 @@ export const calculateImageSizes = ({
       }px" for a resolutions field for
                      the file ${file.absolutePath}
                      was larger than the actual image ${fixedDimension} of ${
-        file[fixedDimension]
+        imgDimensions[fixedDimension]
       }px!
                      If possible, replace the current image with a larger one.
                      `)
     }
-  } else {
+  } else if (layout === `fluid` || layout === `constrained`) {
     // FLUID
     // if no maxWidth is passed, we need to resize the image based on the passed maxHeight
     const fixedDimension = maxWidth === undefined ? `maxHeight` : `maxWidth`
     maxWidth = maxWidth
```

Comment:
This will fail if neither maxHeight or maxWidth is set, as it tries to multiply maxHeight by the aspect ratio. If neither is set, you should set the *width* to the default size and calculate the height from that

Reply:
I _think_ I've handled this based on what I understand. It's not the most functional logic like this writing over values over and over but seems to get the job done: https://github.com/gatsbyjs/gatsby/pull/27554/commits/c399322dfffdac9757897956dbc4e865b70aab83#diff-36db1668fae19fc6f3b88538a610293a66071255a4633e383b76b9b7e79b97edR168-R181